proposals/0026/README.md

```diff
@@ -0,0 +1,346 @@
+
+## Goals and not Goals
+
+Goals:
+* **Compatibility with existing Tink workers, workflow execution, and actions.**
+* **More easily support non-request serving controllers in Tinkerbell.**
+  In this architecture, controllers like PBnJ could leverage Kubernetes primitives like [Custom Resource Definitions][crds] (CRDs), [WATCH APIs][watch], and [Field Management][fm] to complete workflow steps.
+* **Reduce the security surface of the Tinkerbell API.**
+  Tinkerbell is a high-value component of data center infrastructure, so protection of DHCP infrastructure, BMC/IPMI management, needs to be treated as such.
+  Implementing multiple authorization modes is a non-trivial task.
+  The fewer APIs, and authorization options, and lines of code that exist, the fewer opportunities there are for security issues to arise.
+* **Support a highly-available architecture.**
+  Postgres is a fantastic database, but managing high-availability with graceful failover is not trivial.
+  Using an alternative data store that better supports failure will better help operators to have higher availability and not require downtime for upgrades or failover.
+* **Add KRM support for existing components**.
+  Hegel, Boots, and Tinkerbell API will be modified to use Kubernetes as the datastore with a feature flag
+* **Support migration of existing installations.**
+  Migration tooling and documentation will be provided for existing Workflows, Hardware, and Templates stored in Tinkerbell.
+  Migration should be as straightforward as:
+  * Creating Tinkerbell CRDs in the Kubernetes API
+  * Running a provided migration command
+  * Restarting each Tinkerbell component targeting Kubernetes with a feature flag enabled
+
+Non Goals:
+* **Implement attribute-based authorization in Tink API.**
+  This is intentionally descoped from this proposal and can be implemented in a separate proposal.
+* **Require Tinkerbell to be operated as pods inside Kubernetes.**
+  The Kubernetes API will become a dependency of Tinkerbell, but that API could exist in a cloud provider or on-premise.
+* **Require the use of Cluster API ([CAPI][capi]).**
+  The [Tinkerbell CAPI provider][capt] (CAPT) is mentioned in this proposal only for reference of a known Tinkerbell client.
+  Implementation of this proposal will necessitate changes in CAPT, but that is not the core motivation for this proposal.
+* **Implementation of PBnJ as a Kubernetes controller**
+  This can be implemented in a separate proposal.
+* **Make the Tink worker a client of Kubernetes.**
+  Kubernetes doesn't natively support robust attribute-based identities for non-node identities.
+  For example, if the Tink worker were a Kubernetes client, it is not possible with Kubernetes RBAC to ensure that a particular worker can only view or mutate wobernetes does have the [Node Authorizer][norkflows that relate to the worker.
```

Comment:
**wobernetes** (it looks like you lost a few more words than I can guess)

![image](https://user-images.githubusercontent.com/317653/141793239-36a7783e-b43c-4f16-84aa-2ef41d1d081c.png)

Reply:
Oh.. **norkflows** too.. I'm going to guess you were writing this line while falling asleep to some classic 80s/90s TV.

![image](https://user-images.githubusercontent.com/317653/141793474-104c0f08-e985-49f3-8c2b-4228ad5d8565.png)

Reply:
😆

Reply:
You can probably also thank my [ergodox-ez](https://ergodox-ez.com/) learning curve!